controllers/front/NewProductsController.php

```diff
@@ -24,46 +24,44 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
-class NewProductsControllerCore extends FrontController
+use PrestaShop\PrestaShop\Core\Business\Product\Search\ProductSearchQuery;
+use PrestaShop\PrestaShop\Core\Business\Product\Search\SortOrder;
+use PrestaShop\PrestaShop\Adapter\NewProducts\NewProductsProductSearchProvider;
+use PrestaShop\PrestaShop\Adapter\Translator;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+
+class NewProductsControllerCore extends ProductListingFrontController
 {
     public $php_self = 'new-products';
 
     /**
-     * Assign template vars related to page content
-     * @see FrontController::initContent()
+     * Initializes controller
+     *
+     * @see FrontController::init()
+     * @throws PrestaShopException
      */
-    public function initContent()
+    public function init()
     {
-        parent::initContent();
+        parent::init();
 
-        $this->productSort();
+        $this->doProductSearch('catalog/new-products.tpl');
+    }
 
-        // Override default configuration values: cause the new products page must display latest products first.
-        if (!Tools::getIsset('orderway') || !Tools::getIsset('orderby')) {
-            $this->orderBy = 'date_add';
-            $this->orderWay = 'DESC';
-        }
+    protected function getProductSearchQuery()
+    {
+        $query = new ProductSearchQuery;
+        $query
+            ->setQueryType('new-products')
+            ->setSortOrder(new SortOrder('product', 'date_add', 'desc'))
+        ;
+        return $query;
+    }
 
-        $nb_products = (int)Product::getNewProducts(
-            $this->context->language->id,
-            (isset($this->p) ? (int)$this->p - 1 : null),
-            (isset($this->n) ? (int)$this->n : null),
-            true
+    protected function getDefaultProductSearchProvider()
+    {
+        $translator = new Translator(new LegacyContext);
+        return new NewProductsProductSearchProvider(
+            $translator
         );
-
-        $this->pagination($nb_products);
-
-        $products = Product::getNewProducts($this->context->language->id, (int)$this->p - 1, (int)$this->n, false, $this->orderBy, $this->orderWay);
-        $this->addColorsToProductList($products);
-
-        $this->context->smarty->assign(array(
-            'products' => $products,
-            'add_prod_display' => Configuration::get('PS_ATTRIBUTE_CATEGORY_DISPLAY'),
```

Comment:
Ah OK, it determines whether or not the add to cart button is displayed for variants in product listings. Already handled by productpresenter so safe to remove :)